src/datumaro/plugins/data_formats/tf_detection_api/base.py

```diff
@@ -78,45 +101,33 @@ def _parse_labelmap(cls, text):
-        features = {
-            "image/filename": tf.io.FixedLenFeature([], tf.string),
-            "image/source_id": tf.io.FixedLenFeature([], tf.string),
-            "image/height": tf.io.FixedLenFeature([], tf.int64),
-            "image/width": tf.io.FixedLenFeature([], tf.int64),
-            "image/encoded": tf.io.FixedLenFeature([], tf.string),
-            "image/format": tf.io.FixedLenFeature([], tf.string),
-            # use varlen to avoid errors when this field is missing
-            "image/key/sha256": tf.io.VarLenFeature(tf.string),
-            # Object boxes and classes.
-            "image/object/bbox/xmin": tf.io.VarLenFeature(tf.float32),
-            "image/object/bbox/xmax": tf.io.VarLenFeature(tf.float32),
-            "image/object/bbox/ymin": tf.io.VarLenFeature(tf.float32),
-            "image/object/bbox/ymax": tf.io.VarLenFeature(tf.float32),
-            "image/object/class/label": tf.io.VarLenFeature(tf.int64),
-            "image/object/class/text": tf.io.VarLenFeature(tf.string),
-            "image/object/mask": tf.io.VarLenFeature(tf.string),
-        }
+
+        # labelmap_path = osp.join(osp.dirname(filepath), DetectionApiPath.LABELMAP_FILE)
+        files = os.listdir(osp.dirname(filepath))
+        for filename in files:
+            if DetectionApiPath.LABELMAP_FILE in filename:
+                labelmap_path = osp.join(osp.dirname(filepath), filename)
+                break
 
         dataset_labels = OrderedDict()
-        labelmap_path = osp.join(osp.dirname(filepath), DetectionApiPath.LABELMAP_FILE)
         if osp.exists(labelmap_path):
             with open(labelmap_path, "r", encoding="utf-8") as f:
                 labelmap_text = f.read()
             dataset_labels.update(
-                {label: id - 1 for label, id in cls._parse_labelmap(labelmap_text).items()}
+                {label: id - 1 for label, id in self._parse_labelmap(labelmap_text).items()}
             )
 
         dataset_items = []
 
         for record in dataset:
-            parsed_record = tf.io.parse_single_example(record, features)
-            frame_id = parsed_record["image/source_id"].numpy().decode("utf-8")
-            frame_filename = parsed_record["image/filename"].numpy().decode("utf-8")
-            frame_height = tf.cast(parsed_record["image/height"], tf.int64).numpy().item()
-            frame_width = tf.cast(parsed_record["image/width"], tf.int64).numpy().item()
+            parsed_record = tf.io.parse_single_example(record, self._features)
+            frame_id = parsed_record.get("image/source_id", None)
+            frame_id = frame_id.numpy().decode("utf-8") if frame_id else frame_id
+            frame_filename = parsed_record.get("image/filename", None).numpy().decode("utf-8")
+            frame_height = tf.cast(parsed_record.get("image/height", 0), tf.int64).numpy().item()
+            frame_width = tf.cast(parsed_record.get("image/width", 0), tf.int64).numpy().item()
```

Comment:
Do we need these change except `frame_id`? If `parsed_record.get("image/filename", None) => None`, `None.numpy()` is impossible. This type of error can be confusing than raising `KeyError`.

Reply:
Alright, I have thought about it, but `tf_detection_api` is used to contain them. Only the `frame_id` is not included in Roboflow TFRecord format. Is it better to update others under consideration of `KeyError`?

Reply:
I think that it is better to raise `KeyError` in this case because it is genuinely `KeyError`.